python/export_util.py

```diff
@@ -108,6 +113,292 @@ def get_graph(ctx: mgp.ProcCtx) -> List[Union[Node, Relationship]]:
+    new_filename = HEADER_PATH
+    return os.path.join(directory, new_filename)
+
+
+# this will probably be removed, since it is written somewhere else?
+def to_duration_iso_format(value: timedelta) -> str:
+    """Converts timedelta to ISO-8601 duration: P<date>T<time>"""
+    date_parts: List[str] = []
+    time_parts: List[str] = []
+
+    if value.days != 0:
+        date_parts.append(f"{abs(value.days)}D")
+
+    if value.seconds != 0 or value.microseconds != 0:
+        abs_seconds = abs(value.seconds)
+        hours = floor(abs_seconds / 3600)
+        minutes = floor((abs_seconds - hours * 3600) / 60)
+        seconds = abs_seconds - hours * 3600 - minutes * 60
+        microseconds = value.microseconds
+
+        if hours > 0:
+            time_parts.append(f"{hours}H")
+        if minutes > 0:
+            time_parts.append(f"{minutes}M")
+        if seconds > 0 or microseconds > 0:
+            microseconds_part = (
+                f".{abs(value.microseconds)}"
+                if value.microseconds != 0
+                else ""
+            )
+            time_parts.append(f"{seconds}{microseconds_part}S")
+
+    date_duration_str = "".join(date_parts)
+    time_duration_str = f'T{"".join(time_parts)}' if time_parts else ""
+
+    return f"P{date_duration_str}{time_duration_str}"
+
+
+def write_file(path, delimiter, quoting_type, data):
+    with open(path, "w") as file:
+        writer = csv.writer(
+            file, delimiter=delimiter, quoting=quoting_type, escapechar="\\"
+        )
+        writer.writerows(data)
+
+
+@mgp.read_proc
+def csv_graph(
+    nodes_list: mgp.List[mgp.Vertex],
+    relationships_list: mgp.List[mgp.Edge],
+    path: str = "",
+    config: mgp.Map = {},
+) -> mgp.Record(path=str, data=str):
+    """
+    Procedure to export the given graph to a csv file. The graph is given with two lists, one for nodes, and one for relationships.
+
+
+    Parameters
+
+    ----------
+
+    nodes_list : List
+
+        A list containing nodes of the graph
+
+    relationships_list : List
+
+        A list containing relationships of the graph
+
+    path : str
+
+        Path to the JSON file containing the exported graph database.
+
+    config : mgp.Map
+
+        stream (bool) = False: Flag to export the graph data to a stream.
+
+        delimiter (string) = ,: Delimiter for csv file.
+
+        quotes (string) = always : Option which quoting type to use
+
+        separateHeader (bool) = False: Flag to separate header into another csv file
+
+    """
+    if path == "":
+        path = "exported_file.csv"
+    delimiter, quoting_type, separateHeader, stream = csv_process_config(
+        config
+    )
+    (
+        graph,
+        node_properties,
+        relationship_properties,
+    ) = get_graph_info_from_lists(nodes_list, relationships_list)
+    data_list = csv_data_list(graph, node_properties, relationship_properties)
+    header = csv_header(node_properties, relationship_properties)
```

Comment:
Since the `header` variable is only used as `[header]`, maybe change the return type of `csv_header`?